File: 2219-maximum-number-of-words-found-in-sentences/2219-maximum-number-of-words-found-in-sentences.cpp
class Solution {
public:
    int mostWordsFound(vector<string>& sentences) {
        int maxWords = 0;
        
        for (string  sentence : sentences) {
            int spaceCount = 0;
            for (char ch : sentence) {
                if (ch == ' ') {
                    spaceCount++;
                }
            }
            int wordCount = spaceCount + 1;
            if (wordCount > maxWords) {
                maxWords = wordCount;
            }
        }

        return maxWords;
    }
};

File: 2219-maximum-number-of-words-found-in-sentences/2219-maximum-number-of-words-found-in-sentences.java
class Solution {
    public int mostWordsFound(String[] sentences) {
        int max_space_cnt = 0;

        for (int i = 0; i < sentences.length; i++) {
            int space_cnt = 0;

            for (int j = 0; j < sentences[i].length(); j++) {
                if (sentences[i].charAt(j) == ' ')
                    space_cnt++;
            }

            if (space_cnt > max_space_cnt)
                max_space_cnt = space_cnt;
        }

        return max_space_cnt + 1; 
    }
}
